sdk/python/lib/test/provider/experimental/test_analyzer.py

```diff
@@ -0,0 +1,151 @@
+# Copyright 2025, Pulumi Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Optional, TypedDict, Union
+
+import pulumi
+from pulumi.provider.experimental.metadata import Metadata
+from pulumi.provider.experimental.analyzer import Analyzer, unwrap_input, unwrap_output
+from pulumi.provider.experimental.component import (
+    ComponentDefinition,
+    PropertyDefinition,
+    PropertyType,
+)
+
+metadata = Metadata("my-component", "0.0.1")
+
+
+def test_analyze_component():
+    class SelfSignedCertificateArgs(TypedDict):
+        algorithm: pulumi.Input[str]
+        ecdsa_curve: Optional[pulumi.Input[str]]
+        bits: Optional[pulumi.Input[int]]
```

Comment:
Should we add all possible simple types for completeness?

Reply:
There's a `test_analyze_component_plain_types` below